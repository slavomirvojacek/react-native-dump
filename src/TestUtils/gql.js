const assertQueryOrMutation = (QoM) => {
  test('matches Query or Mutation', () => {
    expect(QoM).toMatchSnapshot()
  })
}

const assertGql = (Component) => {
  const { gql } = Component
  expect(gql instanceof Object).toEqual(true)
  Object
    .keys(gql)
    .forEach(k => assertQueryOrMutation(gql[k]))
}

export { assertGql }
