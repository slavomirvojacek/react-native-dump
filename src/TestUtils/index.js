export { assertGql } from './gql'
export { mockSetState } from './mocking'
export { assertSnapshots } from './snapshot'
