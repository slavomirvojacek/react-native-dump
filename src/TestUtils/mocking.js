// Sometimes, we might want to test internal state changes without having to render.
// This overrides the setState method to only update the state, without interacting with
// React's internals such as the scheduler.
const mockSetState = (Component) => {
  return class extends Component {
    setState (f) {
      if (typeof f === 'function') {
        this.state = {...this.state, ...f(this.state)}
      } else {
        this.state = {...this.state, ...f}
      }
    }
  }
}

export { mockSetState }
