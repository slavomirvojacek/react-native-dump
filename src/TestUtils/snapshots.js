import React from 'react'
import { Provider } from 'react-redux'
import renderer from 'react-test-renderer'

const assertSnapshots = (Component, props, desc, store = null) => {
  let getComponent = p => null

  if (store !== null) {
    getComponent = p => (
      <Provider store={store}>
        <Component {...p}/>
      </Provider>
    )
  } else {
    getComponent = p => <Component {...p}/>
  }

  test(desc, () => {
    const tree = renderer
      .create(getComponent(props))
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
}

export { assertSnapshots }
