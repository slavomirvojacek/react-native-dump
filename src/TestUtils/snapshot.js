import React from 'react'
import renderer from 'react-test-renderer'

import { mockSetState } from './mocking'

const assertSnapshot = (C, config) => {
  const {props, state, desc, throws} = config

  if (throws) {
    test(desc, () => {
      expect(() => {
        if (C.prototype.render) {
          new C(props)
        } else {
          C(props)
        }
      }).toThrow(throws)
    })
  } else {
    let Component
    let RenderedComponent

    if (C.prototype.render) {
      const CWithMockState = mockSetState(C)
      Component = new CWithMockState(props)

      if (state) {
        Component.setState(state)
      }

      RenderedComponent = Component.render()
    } else {
      Component = C(props)
      RenderedComponent = Component
    }

    test(desc, () => {
      const tree = renderer
        .create(RenderedComponent)
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
  }
}

const assertSnapshots = (Component, configs) => {
  configs.forEach(config => assertSnapshot(Component, config))
  // if (Component.gql) assertGql(Component)
}

export { assertSnapshots }
