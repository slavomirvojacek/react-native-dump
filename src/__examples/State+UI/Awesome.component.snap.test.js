import React from 'react'

import AwesomeComponent from './Awesome.component'

import { assertSnapshots } from '../../TestUtils/snapshot'

describe('AwesomeComponent Snapshots', () => {
  const config = [
    {
      props: AwesomeComponent.defaultProps,
      desc: 'renders correctly'
    },
    {
      props: {...AwesomeComponent.defaultProps, color: 'red'},
      desc: 'renders correctly when color is red'
    },
    {
      props: {...AwesomeComponent.defaultProps, isWarningVisible: true},
      desc: 'renders correctly when isWarningVisible is true'
    }
  ]

  assertSnapshots(AwesomeComponent, config)
})
