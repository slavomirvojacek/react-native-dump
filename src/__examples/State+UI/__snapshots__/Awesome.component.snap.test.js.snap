// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AwesomeComponent Snapshots matches query 1`] = `undefined`;

exports[`AwesomeComponent Snapshots renders correctly 1`] = `
<View>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    onPress={[Function]}
    style={
      Object {
        "color": "black",
      }
    }
  >
    Click to change the color of this text to blue
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    onPress={[Function]}
  >
    Show odd warning
  </Text>
</View>
`;

exports[`AwesomeComponent Snapshots renders correctly when color is red 1`] = `
<View>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    onPress={[Function]}
    style={
      Object {
        "color": "red",
      }
    }
  >
    Click to change the color of this text to blue
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    onPress={[Function]}
  >
    Show odd warning
  </Text>
</View>
`;

exports[`AwesomeComponent Snapshots renders correctly when isWarningVisible is true 1`] = `
<View>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    onPress={[Function]}
    style={
      Object {
        "color": "black",
      }
    }
  >
    Click to change the color of this text to blue
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    onPress={[Function]}
  >
    Hide this odd warning
  </Text>
</View>
`;
